src/user/user.controller.ts

```diff
@@ -26,16 +29,39 @@ export class UserController {
     }
   }
 
-  @Get()
-  findAll() {
-    return this.userService.findAll();
+  @Get('profile')
+  @UseGuards(JwtAuthGuard)
+  async getProfile(@Req() req: Request & { user: any }) {
+    if (!req.user || !req.body.email) {
+      throw new BadRequestException('Invalid user data.');
+    }
+    return this.userService.findByEmail(req.body.email);
   }
+  
 
-  @Get(':id')
-  findOne(@Param('id') id: string) {
-    return this.userService.findOne(+id);
+//@Get(':id')
+findOne(@Param('id') id: string) {
+  const numericId = Number(id);
+  if (isNaN(numericId) || numericId <= 0) {
+    throw new BadRequestException('The ID must be a positive number.');
+  }
+  return this.userService.findOne(numericId);
+}
+  
+  @Get('all')
+  @UseGuards(JwtAuthGuard)
+  findAll(@Req() req: Request & { user: any }) {
+    console.log(req.user);
+    if (!req.user) {
+      throw new BadRequestException('Invalid user data.');
+    }
+    if (req.user.role !== 'admin') {
```

Comment:
It would be better if you use a guard for checking if the user trying to execute the action is an admin or not

Reply:
fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/22/commits/125b085742a7fb2d6dd51253ab45379a574d1456